src/pages/Lending/Pool/components/Repay/RepayConfirm.tsx

```diff
@@ -121,7 +126,59 @@ export const RepayConfirm = ({
     select: mutation => mutation.state.submittedAt,
   })
 
+  const collateralAsset = useAppSelector(state => selectAssetById(state, collateralAssetId))
+  const repaymentFeeAsset = useAppSelector(state =>
+    selectFeeAssetById(state, repaymentAsset?.assetId ?? ''),
+  )
+  const collateralFeeAsset = useAppSelector(state => selectFeeAssetById(state, collateralAssetId))
+
+  const eventData = useMemo(() => {
+    if (!confirmedQuote) return {}
+
+    const assets = selectAssets(store.getState())
+
+    const compositeRepaymentAsset = getMaybeCompositeAssetSymbol(
+      repaymentAsset?.assetId ?? '',
+      assets,
+    )
+    const compositeCollateralAsset = getMaybeCompositeAssetSymbol(
+      collateralAsset?.assetId ?? '',
+      assets,
+    )
+
+    return {
+      repaymentAsset: compositeRepaymentAsset,
+      collateralAsset: compositeCollateralAsset,
+      repaymentAssetChain: repaymentFeeAsset?.networkName,
+      collateralAssetChain: collateralFeeAsset?.networkName,
+      totalFeesUserCurrency: bn(confirmedQuote.quoteTotalFeesFiatUserCurrency).toFixed(2),
+      totalFeesUsd: bn(confirmedQuote.quoteTotalFeesFiatUsd).toFixed(2),
+      repaymentPercent: confirmedQuote.repaymentPercent,
+      repaymentAmountUserCurrency: bnOrZero(confirmedQuote.repaymentAmountFiatUserCurrency).toFixed(
+        2,
+      ),
+      repaymentAmountUsd: bnOrZero(confirmedQuote.repaymentAmountFiatUsd).toFixed(2),
+      repaymentAmountCryptoPrecision: confirmedQuote.repaymentAmountCryptoPrecision,
+      debtRepaidAmountUserCurrency: bn(confirmedQuote.quoteDebtRepaidAmountUserCurrency).toFixed(2),
+      debtRepaidAmountUsd: bn(confirmedQuote.quoteDebtRepaidAmountUsd).toFixed(2),
+      collateralDecreaseCryptoPrecision: confirmedQuote.quoteLoanCollateralDecreaseCryptoPrecision,
+      collateralDecreaseUserCurrency: bn(
+        confirmedQuote.quoteLoanCollateralDecreaseFiatUserCurrency,
+      ).toFixed(2),
+      collateralDecreaseUsd: bn(confirmedQuote.quoteLoanCollateralDecreaseFiatUsd).toFixed(2),
+    }
+  }, [
+    collateralAsset?.assetId,
+    collateralFeeAsset?.networkName,
+    confirmedQuote,
+    repaymentAsset?.assetId,
+    repaymentFeeAsset?.networkName,
+  ])
+
+  const mixpanel = getMixPanel()
+
```

Comment:
same goes here, might be able to do all of this in a hook

Reply:
See the comment above https://github.com/shapeshift/web/pull/5920#discussion_r1434825650